src/server/adapters/logger/logProvider.ts

```diff
@@ -54,14 +63,68 @@ const getLogDNA = ({ apiKey, environment }: LogDNAParams): LogProvider => {
     };
 };
 
+const getPino = ({ logFilePath, environment }: PinoParams): LogProvider => {
+    const nrPinoConf = nrPino();
+
+    const pinoConf = {
+        ...nrPinoConf,
+        formatters: {
+            ...nrPinoConf.formatters,
+            level: (label: string) => ({ level: label })
+        },
+        base: {
+            service: environment === "prod" ? "microsoft-teams-app" : `microsoft-teams-app-${environment}`
+        }
+    };
+    const stream = logFilePath ? pino.destination(logFilePath) : undefined;
+    if (stream) {
+        /*
+         This is required for pino to keep on writing logfile after log rotation.
+         'SIGHUP' event be triggered by logrotate after log rotation.
+         */
+        process.on("SIGHUP", () => stream.reopen());
+    }
+
+    const logger = stream ? pino(pinoConf, stream) : pino(pinoConf);
+
+    return {
+        info: (message, { meta }): void => logger.info?.({ meta }, message),
+        error: (message, { meta }): void => logger.error?.({ meta }, message),
+        flush: async (): Promise<void> => {
+            logger.flush();
+            await once(logger, "cleared");
+        }
+    };
+};
+
+const getMultipleLogger = ({ apiKey: logDNAApiKey, environment, logFilePath }:MultipleLogParams): LogProvider => {
+    const logdnaLogger = getLogDNA({ apiKey: logDNAApiKey, environment });
+    const pinoLogger = getPino({ environment, logFilePath });
+    return {
+        info: (message:string, { meta }): void => {
+            logdnaLogger.info(message, { meta });
+            pinoLogger.info(message, { meta });
+        },
+        error: (message:string, { meta }): void => {
+            logdnaLogger.error(message, { meta });
+            pinoLogger.error(message, { meta });
+        },
+        flush: async (): Promise<void> => {
+            await logdnaLogger.flush();
+            await pinoLogger.flush();
+        }
+    };
+};
+
 interface LogProviderGetParams {
     logDNAApiKey?: string;
     environment: string;
+    logFilePath: string;
 }
 
-const getLogProvider = ({ logDNAApiKey, environment }: LogProviderGetParams): LogProvider => {
+const getLogProvider = ({ logDNAApiKey, environment, logFilePath }: LogProviderGetParams): LogProvider => {
```

Comment:
Maybe we should return pino instance instead of console, based on environment 🤔

Reply:
with default settings, pino's readability is a bit poor (produces JSON log lines), to make it a bit readable, we need packages like pino-pretty, I didn't put effort into integrating pino-pretty sake of simplicity as we have already console logger here.

Reply:
I meant to initialize pino even if logDNAApiKey is undefined when `NODE_ENV===local`